Assets/__Scripts/UI/SongEditMenu/DifficultyRow.cs

```diff
@@ -0,0 +1,64 @@
+﻿using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class DifficultyRow
+{
+    public Transform Obj { get; private set; }
+    public Image Background { get; private set; }
+    public string Name { get; private set; }
+    public Toggle Toggle { get; private set; }
+    public Button Button { get; private set; }
+    public TMP_InputField NameInput { get; private set; }
+    public Button Copy { get; private set; }
+    public Image CopyImage { get; private set; }
+    public Button Save { get; private set; }
+    public Button Revert { get; private set; }
+    public Button Paste { get; private set; }
+
+    public DifficultyRow(Transform obj)
```

Comment:
Could we perhaps make a `DifficultyRowContainer` script that simply holds references to all of the things we need to grab in this constructor, and just grab stuff from it? I'm not so sure about all this `GetComponent` and `Find` calls, but if it is not being called often then I guess its fine.

Reply:
This is only called once per row (so 5 times total) when the Scene loads and then not again.